src/modules/Packages.rb

```diff
@@ -836,10 +841,25 @@ def Reset(keep)
         end
       end
 
+      # special handling for packages, there might be tens of thousands packages
+      # so we need to do it more effectively
+      if keep.include?(:package)
+        # first query the selected packages, then get the details for each selected
+        # package separately to avoid processing a huge list at once
+        # (true = names only without version)
+        Pkg.GetPackages(:selected, true).each do |pkg|
+          Pkg.ResolvableProperties(pkg, :package, "").each do |resolvable|
+            next if resolvable["status"] != :selected || resolvable["transact_by"] == :solver
```

Comment:
I dislike the filtering logic duplication here. I have an idea to do all this in a different way, by constructing a different enumerator for `resolvables` above, but need to check if it works at all and is not even more complicated than this.

Reply:
I also noticed the duplication but as this is for SP1 I rather prefer a simple although not a perfect fix.

Reply:
OK, here's my experiment with a lazy enumerator: https://gist.github.com/mvidner/551d56d676e5700db6926c480276ee37